pkg/generator/golang_generator_test.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 API Testing Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package generator
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+
+	atest "github.com/linuxsuren/api-testing/pkg/testing"
+)
+
+func TestGolangGenerator_Generate(t *testing.T) {
+	generator := NewGolangGenerator()
+
+	testSuite := &atest.TestSuite{
+		Name: "Test Suite Example",
+	}
+
+	testcase := &atest.TestCase{
+		Name: "Test Case Example",
+		Request: atest.Request{
+			Method: "POST",
+			API:    "http://127.0.0.1:8082/metrics",
+			Header: map[string]string{
+				"Content-Type": "application/json",
+			},
+			Body: atest.RequestBody{Value: ""},
+		},
+	}
+
+	result, err := generator.Generate(testSuite, testcase)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+
+	expectedSubstring := "http.NewRequest"
```

Comment:
imo, we can provide a go file, then use generate go file to compare. more than just checking imported pkgs & some strings

Reply:
+1 from me
Comparing two files is much more clear. And it can avoid some potential small issues.